### ISSUE TITLE
AWS Redshift: The `NONATOMIC` parameter is not supported when creating a stored procedure

### ISSUE THREAD

**Issue by keithhuster** [dbeaver/dbeaver#20524]
### Description

I am attempting to execute a stored procedure creation script that includes the relatively new `NONATOMIC` parameter in the script.
**Reference**: https://aws.amazon.com/about-aws/whats-new/2023/05/amazon-redshift-auto-commit-statements-stored-procedures/
**Reference**: https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_PROCEDURE.html

**Expected Result**
* The script executes successfully and the stored procedure is created.

**Actual Result**
* The following error message is presented and the script fails to execute.
![image](https://github.com/dbeaver/dbeaver/assets/42976208/6abfe890-d8e5-454a-ac7b-dd169d07a7af)

### DBeaver Version

Community Edition 23.1.2.202307091549

### Operating System

Windows 10 Pro

### Database and driver

Database: AWS Redshift
Driver: AWS / Redshift

### Steps to reproduce

1. Write a script to create a stored procedure.
2. Include the `NONATOMIC` parameter in the script definition.
3. Execute the script.
4. Notice the resulting error and the script fails to execute.

### Additional context

DBeaver does not seem to recognize the `NONATOMIC` parameter as it is not syntax-highlighted in the script editor.

**Comment by LonwoLonwo** [comment]
Hello @keithhuster 

Could you please provide the log files? You can find instructions here: https://dbeaver.com/docs/wiki/Log-files/. Thank you.

But for now, I do not think that this is DBeaver error. Are you sure you used the correct Redshift server version during the script execution?

**Comment by keithhuster** [comment]
@LonwoLonwo Thanks for the response. As far as I can tell, I am using the correct JDBC driver for Redshift according to the [AWS Redshift JDBC documentation](https://docs.aws.amazon.com/redshift/latest/mgmt/jdbc20-install-driver.html).

![image](https://github.com/dbeaver/dbeaver/assets/42976208/55170dee-c714-4fae-9d05-12048b202433)

I have also attached the relevant information from the log files per your request.

**File: .log**
```
!ENTRY org.jkiss.dbeaver.model 4 0 2023-07-11 06:07:54.765
!MESSAGE SQL Error [42601]: ERROR: syntax error at or near "NONATOMIC"
  Position: 212
!SUBENTRY 1 org.jkiss.dbeaver.model 4 0 2023-07-11 06:07:54.765
!MESSAGE ERROR: syntax error at or near "NONATOMIC"
  Position: 212
!STACK 0
com.amazon.redshift.util.RedshiftException: ERROR: syntax error at or near "NONATOMIC"
  Position: 212
	at com.amazon.redshift.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2601)
	at com.amazon.redshift.core.v3.QueryExecutorImpl.processResultsOnThread(QueryExecutorImpl.java:2269)
	at com.amazon.redshift.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1880)
	at com.amazon.redshift.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1872)
	at com.amazon.redshift.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:368)
	at com.amazon.redshift.jdbc.RedshiftStatementImpl.executeInternal(RedshiftStatementImpl.java:514)
	at com.amazon.redshift.jdbc.RedshiftStatementImpl.execute(RedshiftStatementImpl.java:435)
	at com.amazon.redshift.jdbc.RedshiftStatementImpl.executeWithFlags(RedshiftStatementImpl.java:376)
	at com.amazon.redshift.jdbc.RedshiftStatementImpl.executeCachedSql(RedshiftStatementImpl.java:362)
	at com.amazon.redshift.jdbc.RedshiftStatementImpl.executeWithFlags(RedshiftStatementImpl.java:339)
	at com.amazon.redshift.jdbc.RedshiftStatementImpl.execute(RedshiftStatementImpl.java:329)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:330)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:582)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$1(SQLQueryJob.java:491)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:189)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:498)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.run(SQLQueryJob.java:238)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**File: dbeaver-debug.log**
Note: I redacted user and server info in this log file.
[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/12014566/dbeaver-debug.log)

**Comment by LonwoLonwo** [comment]
It is not enough to have the correct driver version. What about your server version? Could you please provide info from the "Test connection" button in your Connection settings?

**Comment by keithhuster** [comment]
@LonwoLonwo Here is the info from the Test Connection button.

![image](https://github.com/dbeaver/dbeaver/assets/42976208/de0e36f5-ea26-4252-9287-9491b62f2a84)

**Comment by LonwoLonwo** [comment]
Ok, could you also provide your stored procedure query?

**Comment by keithhuster** [comment]
Here is the stored procedure signature with the inner details redacted:

```SQL
CREATE OR REPLACE PROCEDURE public.my_stored_proc(
    IN param1 INTEGER
    , IN param2 INTEGER
    , IN param3 FLOAT
    )
    LANGUAGE plpgsql
    NONATOMIC
AS $$

DECLARE
    -- Variable declarations here.
BEGIN
    -- Stored proc implementation here.
END;

$$;
```

**Comment by LonwoLonwo** [comment]
Did you try to highlight the whole procedure text and then click on the execution in the SQL Editor?

**Comment by keithhuster** [comment]
I had been running the script using the **Execute SQL script** button. This resulted in the original error specified in this issue.

![image](https://github.com/dbeaver/dbeaver/assets/42976208/eba61107-197c-495d-a37b-205952a87422)

I tested your suggestion of highlighting the entire script then running the script via the **Execute SQL query** button. This also resulted in the same syntax error due to the `NONATOMIC` keyword.

![image](https://github.com/dbeaver/dbeaver/assets/42976208/d2effd2c-ee92-4430-80bf-80b705e793ca)

![image](https://github.com/dbeaver/dbeaver/assets/42976208/be8e4595-ce3b-4fc4-b106-383652f1e7c0)

**Comment by LonwoLonwo** [comment]
Do you have the ability to check your query in another place? AWS console or something?
I see that your Redshift server version is new. As well as the driver version.
But still, the error is not from DBeaver, but from Redshift:
`at com.amazon.redshift.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2601)`

**Comment by keithhuster** [comment]
I have opened a support ticket with AWS regarding the use of the `NONATOMIC` keyword. I will update this issue once I receive feedback from them.

**Comment by E1izabeth** [comment]
It's been a while since no update here. If the issue is still actual please let me know and provide additional information.

**Comment by keithhuster** [comment]
@E1izabeth I was able to resolve this error thanks to AWS support. The `NONATOMIC` keyword must be specified prior to the `LANGUAGE` keyword. An example working stored procedure definition is shown below.

```SQL
CREATE OR REPLACE PROCEDURE public.my_stored_proc(
    IN param1 INTEGER
    , IN param2 INTEGER
    , IN param3 FLOAT
    )
    NONATOMIC
    LANGUAGE plpgsql
AS $$

DECLARE
    -- Variable declarations here.
BEGIN
    -- Stored proc implementation here.
END;

$$;
```

However, when using the DBeaver "View Function" utility, the displayed stored procedure source code does not show the `NONATOMIC` keyword even though it is actually there. I confirmed this by executing a `SHOW PROCEDURE my_stored_proc(int4, int4, float8);` command which did report the `NONATOMIC` keyword as being present. I will open a new issue for this.
